Angular13/Angulardemo/src/app/login/login.component.spec.ts

```diff
@@ -29,7 +33,9 @@ describe('LoginComponent', () => {
       ],
       declarations: [LoginComponent],
       providers: [
-        { provide: Router, useValue: router }
+        { provide: Router, useValue: router },
+        HttpClientModule,
```

Comment:
no need to add in providers... do like imports: [HttpClientTestingModule]
//always use "HttpClientTestingModule" in spec dont use "HttpClientModule"